src/veh_interact.cpp

```diff
@@ -2247,12 +2311,43 @@ void veh_interact::complete_vehicle()
         break;
     }
 
-    case 'f':
-        if (!g->pl_refill_vehicle(*veh, vehicle_part, true)) {
-            debugmsg ("complete_vehicle refill broken");
+    case 'f': {
+        if( g->u.activity.targets.size() < 1 || !g->u.activity.targets.front() ) {
+            debugmsg( "Activity ACT_VEHICLE: missing refill source" );
+            break;
+        }
+
+        auto &src = g->u.activity.targets.front();
+
+        auto &pt = veh->parts[ vehicle_part ];
+        if( pt.is_tank() && src->is_watertight_container() && !src->contents.empty() ) {
+
+            int qty = std::min( src->contents.front().charges, pt.ammo_capacity() - pt.ammo_remaining() );
+            pt.ammo_set( src->contents.front().typeId(), pt.ammo_remaining() + qty );
```

Comment:
This snippet reminds me `vehicle::refill()` and, to a less extent, `item::fill_with()`. Can it be unified somehow or/and, at least, moved out of the UI function?